### PR TITLE
Add analytics 1

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently, there is no Google Analytics, Facebook Pixel or Hotjar implemented.

**What is the new behavior (if this is a feature change)?**
- Google Analytics, Facebook Pixel and Hotjar has been implemented.

**Other information**:
- The version has been updated to v1.5.1.

### CODE REVIEW

**Review by LucaAllmaier — APPROVED** [review]
Looks good to me.

**Review by Aichnerc — COMMENTED** [review]
Looking good to my old eyes.

**Review by sweggerat — APPROVED** [review]
looks good

File: src/App.js [hunk 1]
```diff
@@ -41,6 +47,99 @@ class App extends React.Component {
     darkMode: mode,
   };
 
+  componentDidMount = () => {
+    this.checkCookies();
+  };
+
+  checkCookies = () => {
+    // Create custom user id for tracking
+    let userId = localStorage.getItem("userId");
```
Comment (pinterid) [inline src/App.js:56]: Add newline

File: src/App.js [hunk 2]
```diff
@@ -41,6 +47,99 @@ class App extends React.Component {
...
+    if (!userId) {
+      const sha256 = require("js-sha256");
+      userId = sha256.create();
+      localStorage.setItem("userId", userId);
+    }
+
+    // Check cookies
+    let cookie = localStorage.getItem("cookie");
```
Comment (pinterid) [inline src/App.js:66]: Add newline.

File: src/App.js [hunk 3]
```diff
@@ -41,6 +47,99 @@ class App extends React.Component {
...
+    if (cookie) {
+      cookie = JSON.parse(cookie);
```
Comment (pinterid) [inline src/App.js:69]: Add newline.

File: src/App.js [hunk 4]
```diff
@@ -41,6 +47,99 @@ class App extends React.Component {
...
+  componentDidMount = () => {
+    this.checkCookies();
+  };
+
+  checkCookies = () => {
+    // Create custom user id for tracking
+    let userId = localStorage.getItem("userId");
+    if (!userId) {
+      const sha256 = require("js-sha256");
+      userId = sha256.create();
```
Comment (pinterid) [inline src/App.js:61]: Add newline

File: src/components/organisms/sections/ContactForm/index.jsx [hunk 5]
```diff
@@ -70,6 +84,9 @@ class ContactForm extends React.Component {
   handleSubmit = (e) => {
     e.preventDefault();
 
+    // Analytics
+    this.props.googleAnalytics.registerContactSend();
+
```
Comment (schettn) [inline src/components/organisms/sections/ContactForm/index.jsx]: It this newline wanted? In my opinion there is no reason to separate those two function calls.

File: src/components/organisms/sections/Hero/index.jsx [hunk 6]
```diff
@@ -160,7 +163,10 @@ class HomePage extends React.Component {
                   aktuellen Stand Deiner gesamten Online-Präsenz und liefert
```
Comment (pinterid) [inline src/components/organisms/sections/Hero/index.jsx]: I would add a "Dir" after "liefert".

File: src/config/fbInit.js [hunk 7]
```diff
@@ -9,10 +9,13 @@ import config from "./fbConfig.js";
 
 // Init Firebase
 firebase.initializeApp(config);
-// Init analytics
-if (config.measurementId) {
+/**
+ * Initialize Google Analytics
+ * Once we implement it ourselves, this is not required for now
+ */
+/*if (config.measurementId) {
```
Comment (schettn) [inline src/config/fbInit.js]: I would recommend to comment out this section using //

**Review by pinterid — CHANGES_REQUESTED** [review]
Looks good after some changes.

File: src/config/fbInit.js [hunk 8]
```diff
@@ -9,10 +9,13 @@ import config from "./fbConfig.js";
 
 // Init Firebase
 firebase.initializeApp(config);
-// Init analytics
-if (config.measurementId) {
+/**
```
Comment (schettn) [inline src/config/fbInit.js]: Is it wanted that you use /** */ as header? Because every other header starts with //>.

**Review by schettn — CHANGES_REQUESTED** [review]
Please check.

File: src/config/fbInit.js (hunk 7 above)
Comment (Aichnerc) [reply src/config/fbInit.js]: I just removed the entire thing. I won't be using it anymore.

**Review by schettn — APPROVED** [review]
🚀

**Review by pinterid — APPROVED** [review]
Looks good to merge.

**Pull request merged**